tools/serverpod_cli/lib/src/analyzer/models/validation/restrictions.dart

```diff
@@ -1185,4 +1204,23 @@ class Restrictions {
 
     return classDefinition.tableName != null;
   }
+
+  List<ClassDefinition> _extractAllClassDefinitionsFromType(
+    TypeDefinition fieldType,
+  ) {
+    var classDefinitions = <ClassDefinition>[];
+
+    if (fieldType.generics.isNotEmpty) {
+      for (var generic in fieldType.generics) {
+        classDefinitions.addAll(_extractAllClassDefinitionsFromType(generic));
+      }
+    }
+
+    var className = fieldType.className;
+    var definition = parsedModels.findByClassName(className);
+
+    if (definition is ClassDefinition) classDefinitions.add(definition);
+
+    return classDefinitions;
```

Comment:
Suggestion: (none blocking) We could change this to a Map or a Set to only get the unique types then we would never print the same type error more than once.

Reply:
Good suggestion! Changed it to a Set.